### PR TITLE
MOTOR-902 Since Python 3.10 asyncio.get_event_loop() is deprecated

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by blink1073** [comment]
Updated the tutorial to use `client.get_io_loop()` because there is no running loop yet until we run some async code.

File: motor/core.py
Comment (ShaneHarvey) [inline motor/core.py:1057]: change_stream.close() is an async method so calling it here leaks a coroutine. It's fine for this PR since it's a preexisting issue but could you file a new issue to improve this code example? Really it should work via task cancellation, not through KeyboardInterrupt.

File: pyproject.toml
Comment (ShaneHarvey) [inline pyproject.toml:2]: Why do we need setuptools>=59.0? And should we add "wheel"?

File: tox.ini
Comment (ShaneHarvey) [inline tox.ini:74]: Do we still need to ignore this now that we use setuptools on >=3.10?

File: motor/frameworks/asyncio/__init__.py
Comment (ShaneHarvey) [inline motor/frameworks/asyncio/__init__.py:44]: I was expecting `asyncio.set_event_loop(asyncio.new_event_loop())`. Doesn't `AbstractEventLoopPolicy.get_event_loop()` have the same behavior as asyncio.get_event_loop()?

Did the maintainers just forget to deprecate AbstractEventLoopPolicy.get_event_loop?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: motor/core.py
Comment (ShaneHarvey) [inline motor/core.py:154]: we might need to make a similar change in AgnosticClientEncryption.__init__.

File: tox.ini
Comment (blink1073) [reply tox.ini:74]: The warning is still raised by [pip itself](https://github.com/pypa/pip/blob/c247ddce123513198858d4589c5ee198dcc6ba40/src/pip/_internal/locations/_distutils.py#L9)

File: motor/frameworks/asyncio/__init__.py
Comment (blink1073) [reply motor/frameworks/asyncio/__init__.py:44]: Yes, it hasn't been deprecated yet.  This is the workaround that tornado is using for their 6.2 release.

File: motor/core.py
Comment (blink1073) [reply motor/core.py:154]: Done

**Comment by blink1073** [comment]
We should run the synchro tests here after merging #158 to make sure there aren't any regressions.

File: motor/core.py
Comment (ShaneHarvey) [reply motor/core.py:1057]: I opened https://jira.mongodb.org/browse/MOTOR-938

File: motor/frameworks/asyncio/__init__.py
Comment (ShaneHarvey) [reply motor/frameworks/asyncio/__init__.py:44]: Oh I see, `asyncio.set_event_loop(asyncio.new_event_loop())` will incorrectly reinitialize the loop when there is a loop already set on the thread, but it just hasn't been started yet:
```
>>> asyncio.get_running_loop()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: no running event loop
>>> id(asyncio.get_event_loop())
4361557568
>>> asyncio.set_event_loop(asyncio.new_event_loop())
>>> id(asyncio.get_event_loop())
4374658448
```

File: motor/core.py
Comment (blink1073) [reply motor/core.py:1057]: Thanks, sorry I meant to return to this comment and missed it in the scroll

File: pyproject.toml
Comment (blink1073) [reply pyproject.toml]: We don't need this file at all, I had added it to try to avoid the distutils deprecation warning, removing it

**Review by ShaneHarvey — APPROVED** [review]

File: motor/frameworks/asyncio/__init__.py
Comment (graingert) [reply motor/frameworks/asyncio/__init__.py:44]: @blink1073 this workaround in tornado is only used for the deprecated IOLoop().start() and it is only intended to support

```
with warnings.catch_warnings():
    warnings.simplefilter("ignore")
    loop = IOLoop()
    
...
loop.start()
```

Comment (graingert) [reply motor/frameworks/asyncio/__init__.py:44]: anywhere the user creates an IOLoop still happens via `asyncio.get_event_loop()` so if the user creates an IOLoop too early they get a DeprecationWarning

https://github.com/tornadoweb/tornado/blob/713f06b115f7125b4cf0d21d667ab5c801318baf/tornado/platform/asyncio.py#L295-L297

**Pull request merged**
